In [7]:
import pandas as pd

In [8]:
import numpy as np

In [9]:
from sklearn import tree

In [10]:
from sklearn import preprocessing

In [11]:
titanic_train=pd.read_csv("train.csv")

In [12]:
titanic_train["Age"].mean()


32.69985376827896

In [13]:
meanage=np.where(titanic_train["Age"].isnull(),32,titanic_train["Age"])

In [14]:
titanic_train["Age"]=meanage

In [15]:
label_encoder=preprocessing.LabelEncoder()

In [16]:
encode_sex=label_encoder.fit_transform(titanic_train["Sex"])

In [17]:
tree_model=tree.DecisionTreeClassifier()

In [18]:
predictors=pd.DataFrame([encode_sex,titanic_train["Age"],titanic_train["Fare"]]).T

In [19]:
tree_model.fit(X=predictors,y=titanic_train["Survived"])

DecisionTreeClassifier()

In [20]:
with open("Titanic.dot",'w') as f:
    f=tree.export_graphviz(tree_model,feature_names=["Sex","Age","Fare"],out_file=f)

In [21]:
tree_model.score(X=predictors,y=titanic_train["Survived"])

0.9763779527559056

# The model is 97.6% Accurate and Hence can be used to decide which all pasengers survived and which did not¶


In [22]:
from sklearn.tree import export_text

In [23]:
feature=["Sex","Age","Fare"]
r=export_text(tree_model,feature_names=feature)
print(r)

|--- Sex <= 0.50
|   |--- Fare <= 48.20
|   |   |--- Fare <= 27.82
|   |   |   |--- Fare <= 25.70
|   |   |   |   |--- Fare <= 24.07
|   |   |   |   |   |--- Fare <= 10.48
|   |   |   |   |   |   |--- Fare <= 8.04
|   |   |   |   |   |   |   |--- Fare <= 6.99
|   |   |   |   |   |   |   |   |--- class: 0
|   |   |   |   |   |   |   |--- Fare >  6.99
|   |   |   |   |   |   |   |   |--- Fare <= 7.52
|   |   |   |   |   |   |   |   |   |--- class: 1
|   |   |   |   |   |   |   |   |--- Fare >  7.52
|   |   |   |   |   |   |   |   |   |--- Age <= 14.50
|   |   |   |   |   |   |   |   |   |   |--- class: 0
|   |   |   |   |   |   |   |   |   |--- Age >  14.50
|   |   |   |   |   |   |   |   |   |   |--- Age <= 17.50
|   |   |   |   |   |   |   |   |   |   |   |--- class: 1
|   |   |   |   |   |   |   |   |   |   |--- Age >  17.50
|   |   |   |   |   |   |   |   |   |   |   |--- truncated branch of depth 11
|   |   |   |   |   |   |--- Fare >  8.04
|   |   |   |   |   |   |   |--- Age <= 19